pkg/catalog/catalog.go

```diff
@@ -129,8 +131,67 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
 		imageStreamTagMap[imageStreamTag.Name] = imageStreamTag
 	}
 
-	var builderImages []CatalogImage
+	builderImages := getBuildersFromImageStreams(imageStreams, imageStreamTagMap)
+
+	return builderImages, nil
+}
 
+// SpliceSupportedTags splits the tags in to fully supported and unsupported tags
+func SpliceSupportedTags(catalogImage CatalogImage) ([]string, []string) {
+
+	var supTag, nonSupTag []string
+	tagMap := createImageMap(catalogImage.imageStreamRef.Spec.Tags)
+	for _, tag := range catalogImage.NonHiddenTags {
+		imageName := tagMap[tag]
+		if isSupportedImage(imageName) {
+			supTag = append(supTag, tag)
+		} else {
+			nonSupTag = append(nonSupTag, tag)
+		}
+	}
+	return supTag, nonSupTag
+}
+
+func createImageMap(tagRefs []imagev1.TagReference) map[string]string {
```

Comment:
I know it's not an exported function but wouldn't it be helpful to have a comment that briefs what this function does?

Reply:
added